### DISCUSSION TITLE
One replica of metrics server is not enough.

### DISCUSSION THREAD

**Discussion by ravidbro** [kedacore/keda#3386] [Q&A / Need Help]
Hi

We have big clusters and a lot of ScaledObjects that are pulling frequently, and we have errors due to a single node limitation.
If the ClusterIP of the service resource is defined as the metrics endpoint, is there a reason why we shouldn't set more than one replica to the keda-metrics-server?

**Comment by zroubalik** [comment]
You could have more replicas, but based on your k8s setup the trafic might be routed to just one replica.

Btw what KEDA version are you using?

**Reply by timown** [reply]
> Btw what KEDA version are you using?

2.7.1

**Reply by ravidbro** [reply]
I see that the chart hardcoded one replica, is there a way to use more without forking the chart?
https://github.com/kedacore/charts/blob/main/keda/templates/22-metrics-deployment.yaml#L11

**Reply by zroubalik** [reply]
I think that it is fine to send a PR to address this.

**Reply by amirschw** [reply]
> based on your k8s setup the trafic might be routed to just one replica.

Is this because the HPA controller will try to keep the same connection for every sync with the metric server?

**Reply by ravidbro** [reply]
Does the metric server poll for the metric from the external resource only when HPA send a request to get metric value, or is there a timer that every metric-server pod will poll a metric every 15 seconds regardless of if someone (HPA) asked for the metric?
If the case is the latter, then adding more replicas won't help us since we are trying to reduce the number of calls that each metric-server is making and not to balance calls from HPA to metrics-server which is not our problem now.

**Reply by tomkerkhove** [reply]
I created https://github.com/kedacore/charts/pull/295

**Reply by ravidbro** [reply]
Thanks!
Do you know when the next release is scheduled?

**Reply by tomkerkhove** [reply]
Next week! 🎉

**Reply by tomkerkhove** [reply]
It's out - https://github.com/kedacore/charts/releases/tag/v2.8.0
